### PR TITLE
Make installation script respect $XDG_DATA_HOME

### PR DESCRIPTION
**Pull request by jonz94**
Related to #357 and #613

### CODE REVIEW

File: .ci/install.sh [hunk 1]
```diff
@@ -2,7 +2,12 @@
 
 set -e
 
-INSTALL_DIR="$HOME/.fnm"
+if [ -d "$HOME/.fnm" ]; then
+  INSTALL_DIR="$HOME/.fnm"
+else
+  INSTALL_DIR="${XDG_DATA_HOME:-$HOME/.local/share}/fnm"
```
Comment (Schniz) [inline .ci/install.sh:8]: I wonder if that's enough. MacOS way would be `~/Library/Application Support/fnm`

what do you think? SHould we keep `~/.local/share/fnm` just for the installation process?

Comment (jonz94) [reply .ci/install.sh:8]: Oops, thanks for your friendly reminder.

I just thought on MacOS `fnm` would be installed at `~/.local/share/fnm`

File: .ci/install.sh [hunk 2]
```diff
@@ -2,10 +2,19 @@
 
 set -e
 
-INSTALL_DIR="$HOME/.fnm"
 RELEASE="latest"
 OS="$(uname -s)"
 
+if [ -d "$HOME/.fnm" ]; then
+  INSTALL_DIR="$HOME/.fnm"
+elif [ ! -z $XDG_DATA_HOME ]; then
```
Comment (page-down) [inline .ci/install.sh:10]: You can use `test -n` and double quotes.
```bash
elif [ -n "$XDG_DATA_HOME" ]; then
```

File: .ci/install.sh [hunk 3]
```diff
@@ -2,10 +2,19 @@
...
+  INSTALL_DIR="$XDG_DATA_HOME/fnm"
+elif [ "$OS" = "Darwin" ]; then
+  INSTALL_DIR="$HOME/Library/Application\ Support/fnm"
```
Comment (page-down) [inline .ci/install.sh:13]: You need to remove the extra `\`, because double quotes are already used. Otherwise the path is wrong.

**Review by page-down — COMMENTED** [review]
The path under macOS is wrong and needs to be fixed.

File: .ci/install.sh (hunk 2 above)
Comment (jonz94) [reply .ci/install.sh:10]: Thanks for the tips!

I did try `test -n` before, but I tried it without double quotes:

```
elif [ -n $XDG_DATA_HOME ]; then
```

and in some edge case (e.g. `XDG_DATA_HOME=""`), this will cause unexpected result.

Comment (jonz94) [reply .ci/install.sh:10]: Now I understand that string wrapped by double quotes is very different that without double quotes in bash,
and it's important to know when to wrap string by double quotes or not.

File: .ci/install.sh (hunk 3 above)
Comment (jonz94) [reply .ci/install.sh:13]: Thanks for pointing out my mistake.

File: .ci/install.sh (hunk 1 above)
Comment (ljharb) [reply .ci/install.sh]: There’s no convention on Macs for ~/.local to even exist at all/

**Comment by sarmong** [comment]
Will this be ever merged?

**Comment by Schniz** [comment]
I'm gonna update this PR so it'll run CI

**Comment by page-down** [comment]
@Schniz
The PR has been merged, however XDG_DATA_HOME is still not respected under macOS.
```text
https://github.com/Schniz/fnm/issues/622
```

This results in fnm being installed to `~/.local/share/fnm`, however the `FNM_DIR` given by `fnm env` is still `~/Library/Application Support/fnm`.
Since we already have `src/directories.rs`, it seems relatively straightforward.
Do you have any other concerns?
